### DISCUSSION TITLE
[QUESTION]: Version 3: How custom worker will acknowledge to conductor server if it's reading (executing) a task

### DISCUSSION THREAD

**Discussion by bhawani021** [Netflix/conductor#2258] [Q&A] [answered]
In conductor version 3, `API POST /tasks/{taskId}/ack` has been removed.

https://github.com/Netflix/conductor/blob/main/CHANGELOG.md

How worker can acknowledge to Conductor server about execution of task? Is there any new API that worker need to use to acknowledge Connector servicer.

**Comment by aravindanr** [comment, marked as answer]
There is no need to acknowledge that task execution has begun. Task state changes to `IN_PROGRESS` when `/poll` returns. The worker simply needs to call `/updateTask` with the desired status for the task. `ack` was necessary when the default Worker implementation in the client maintained its own in-memory queue. The client no longer has its own queue and `ack` was deprecated in `2.x` and removed in `3.x`.

**Reply by bhawani021** [reply]
@aravindanr 
I don't see any API `/updateTask` in swagger document for version 3.
Do you mean `POST/api​/tasks`?

**Reply by apanicker-nflx** [reply]
Yes, that is correct. The API is `POST /api/tasks` which is an updateTask operation.
